control/control.c

```diff
@@ -306,6 +313,39 @@ int main(int argc, char *argv[])
 		msg.guestos_rootcert.data = ca_cert;
 		goto send_message;
 	}
+	if (!strcasecmp(command, "pull_csr")) {
+		// need exactly one more argument (certificate file)
+		if (optind != argc-1)
+			print_usage(argv[0]);
+
+		has_response = true;
+		msg.command = CONTROLLER_TO_DAEMON__COMMAND__PULL_DEVICE_CSR;
+		goto send_message;
+	}
+	if (!strcasecmp(command, "push_cert")) {
+		// need exactly one more argument (certificate file)
+		if (optind != argc-1)
+			print_usage(argv[0]);
+
+		const char *dev_cert_file = argv[optind++];
+		off_t dev_cert_len = file_size(dev_cert_file);
+		if (dev_cert_len < 0) {
+			ERROR("Error accessing certificate file %s.", dev_cert_file);
+			exit(-2);
+		}
+		uint8_t *dev_cert = mem_alloc(dev_cert_len);
+		if (file_read(dev_cert_file, (char*)dev_cert, dev_cert_len) < 0) {
+			ERROR("Error reading %s.", dev_cert_file);
```

Comment:
free dev_cert in case of error

Reply:
new commit which does FATAL instead of ERROR and than exit